default/scripting/fields/FLD_CORROSIVE_NANITE_SWARM.focs.txt

```diff
@@ -0,0 +1,66 @@
+FieldType
+    name = "FLD_CORROSIVE_NANITE_SWARM"
+    description = "FLD_NANITE_SWARM_DESC"
+    stealth = 0.01
+    tags = [ "EXOTIC" ]
+    effectsgroups = [
+        EffectsGroup    // move around
+            scope = Source
+            effects = [
+                SetSpeed value = 10
+                MoveTowards speed = Source.Speed/2.8 x = UniverseCentreX y = UniverseCentreY
+                MoveInOrbit speed = Source.Speed/1.4 x = UniverseCentreX y = UniverseCentreY
+            ]
+
+        EffectsGroup    // grow size when young
+            scope = Source
+            activation = And [
+                (Source.Age <= max((UniverseWidth ^ 1.1) / 50, 20))
+                Size high = 50
+            ]
+            effects = SetSize value = Value + min(max(Value * RandomNumber(0.05, 0.1), 1.0), 3.0)
+
+        EffectsGroup    // shrink size when old
+            scope = Source
+            activation = (Source.Age >= max((UniverseWidth ^ 1.1) / 50, 20))
+            effects = SetSize value = Value - min(max(Value * RandomNumber(0.05, 0.1), 1.0), 3.0)
+
+        EffectsGroup    // decrease ship structure
+            scope = And [
+                Ship
+                WithinDistance distance = Source.Size * 0.9 condition = Source
+                Not Monster
+            ]
+            stackinggroup = "NANITE_SWARM_SHIP_REPAIR"
+            effects = SetStructure value = Value - (NamedReal name = "FLD_NANITE_SWARM_STRUCTURE_FLAT" value = 5 * [[SHIP_STRUCTURE_FACTOR]]) /
+                max(1.0, Statistic Count condition = And [
+                    Ship
+                    WithinDistance distance = Source.Size * 0.9 condition = Source
+                    Not Monster
+                ])
+
+        EffectsGroup    // shrink when ships are in...
+            scope = Source
+            activation = Number low = 1 condition = And [
+                Ship
+                WithinDistance distance = Source.Size * 0.9 condition = Source
+                Not Monster
+            ]
+            effects = SetSize value = Value - Statistic Count condition = And [
```

Comment:
I'm not sure if this whole script is just for demo purposes, but I would have the swarm grow rather than shrink when eating...

Reply:
Right, scripting is not finalized, @st-pa-fo probably has some plans?

Reply:
yes well nothing tangible yet, the colour image variant would be great as a stand alone, but feel free to do the scripting as well ^..^ i probably would just have either copy/pasted existing nanites or made them parametrized for the effect like two variables one for the affected meter and one for the inflicted value of affection - i.e. the hull plus for the existing variant, and then hull minus for a new one, and maybe one that eats only fighters out of the bays and one other that rebuilds them. lame. i know. but i had no time to think about it much

Reply:
maybe flesh eating nanites/microbes? only affect organic hulls and monsters. or maybe like the nanites from red dwarf where they resurrect dead populations or embiggen small hull space ships to huge hulls .. and that emotion kraken from the JMCS Esperanto i believe it was called, that induces hallucinations and lets ..
stop. off topic. thanks for asking. i'll get back to that.